hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/BlockManagerImpl.java

```diff
@@ -222,9 +224,7 @@ public AllocatedBlock allocateBlock(final long size, ReplicationType type,
       }
 
       if (null == pipeline) {
-        // TODO: #CLUTIL Make the selection policy driven.
-        pipeline = availablePipelines
-            .get((int) (Math.random() * availablePipelines.size()));
+        pipeline = pipelineChoosePolicy.choosePipeline(availablePipelines);
```

Comment:
We could get a runtime exception from a plugged in policy. Can we have to fallback to default (Random) on error after logging?

Reply:
Done.